Source/Public/LevelStatsCollector.h

```diff
@@ -8,6 +8,79 @@ UCLASS()
 class MAPMETRICSGENERATION_API ALevelStatsCollector final : public AActor
 {
     GENERATED_BODY()
+
 public:
+    ALevelStatsCollector();
+
     void PostInitializeComponents() override;
+    void BeginPlay() override;
+    void Tick( float DeltaTime ) override;
+
+private:
+    void InitializeGrid();
+    void SetupSceneCapture() const;
+    bool ProcessNextCell();
+    void CaptureCurrentView();
+    TOptional< FVector > TraceGroundPosition( const FVector & start_location ) const;
+    void CalculateGridBounds();
+    FString GenerateFileName() const;
+    void LogGridInfo() const;
+
+public:
```

Comment:
all those should be private too (and no need for UPROPERTY)

Reply:
done